### ISSUE TITLE
Structured logging of FMU variables 

### ISSUE THREAD

**Issue by traversaro** [robotology/gazebo-fmi#51]
The need to reliably log the state of the FMU simulated emerged several times. A prototype of this functionality (still missing critical pieces  of the infrastructure) is available in https://github.com/traversaro/gazebo-fmi-1/commit/de31703e79c0a6eca5c4ff12f8cc6c9f1b804ea4#diff-87879b445e0fd2c9203e3f98b65c619dR65 . 

Basic ideas: 
* Any FMU that wants its variable to be logged need to have an option turned on (that by default is off)
* Only one file will be produced for the whole simulation, and the name of the variables in it should be  properly scoped to avoid conflicts (should be done in https://github.com/traversaro/gazebo-fmi-1/commit/de31703e79c0a6eca5c4ff12f8cc6c9f1b804ea4#diff-87879b445e0fd2c9203e3f98b65c619dR65 the `writeToFile` method of `FMULoggerSingleton` 
* Possible file formats to produce could be `.mat` files or `.csv` .

**Comment by traversaro** [comment]
Expanding a bit the FMICoSimulation class, it should be possible to easily also dump any kind of internal state of the FMU.

**Comment by traversaro** [comment]
cc @prashanthr05 @triccyx

**Comment by traversaro** [comment]
Just to clarify: I do not know when/if I will have time to work on the prototype, so feel free to go on and complete it yourself @prashanthr05 @triccyx .

**Comment by triccyx** [comment]
I will be happy to work with this and other related issues ... after the 21 of February!

**Comment by traversaro** [comment]
Yes, I meant just take the code and customize it locally if it is useful for the debugging.

**Comment by traversaro** [comment]
This would be something that could be implemented with the logging library we were discussing about the other day @GiulioRomualdi @diegoferigo @prashanthr05 .

**Comment by triccyx** [comment]
Ok

**Comment by traversaro** [comment]
@triccyx Have you checked the interface defined in https://github.com/traversaro/gazebo-fmi-1/commit/de31703e79c0a6eca5c4ff12f8cc6c9f1b804ea4#diff-87879b445e0fd2c9203e3f98b65c619dR65 ?

**Comment by triccyx** [comment]
No, I will check.

**Comment by traversaro** [comment]
For this it may be worth to check the https://github.com/ADVRHumanoids/MatLogger2 library.
